File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        prefix_sum = [0] * n
        suffix_sum = [0] * n

        # Build prefix sums
        for i in range(1, n):
            prefix_sum[i] = prefix_sum[i - 1] + nums[i - 1]

        # Build suffix sums
        for i in range(n - 2, -1, -1):
            suffix_sum[i] = suffix_sum[i + 1] + nums[i + 1]

        # Find the pivot index
        for i in range(n):
            if prefix_sum[i] == suffix_sum[i]:
                return i

        return -1

File: 0724-find-pivot-index/0724-find-pivot-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
    const n = nums.length;
    const prefixSum = Array(n).fill(0);
    const suffixSum = Array(n).fill(0);

    for(let i = 1; i < n; i++){
        prefixSum[i] = prefixSum[i-1] + nums[i-1] 
    }

    for(let i = n - 2; i >= 0; i--){
        suffixSum[i] = suffixSum[i+1] + nums[i+1] 
    }

    for(let i = 0; i < n; i++){
        if(prefixSum[i]=== suffixSum[i]){
            return i;
        } 
    }

    return -1;



    
};